certbot-apache/certbot_apache/_internal/assertions.py

```diff
@@ -123,8 +123,7 @@ def isEqualVirtualHost(first, second):
         first.addrs == second.addrs and
         first.ssl == second.ssl and
         first.enabled == second.enabled and
-        first.modmacro == second.modmacro and
-        first.ancestor == second.ancestor
```

Comment:
Actually, thinking about this more... the most important check that calls this is here: https://github.com/certbot/certbot/blob/apache-parser-v2/certbot-apache/certbot_apache/_internal/configurator.py#L916

Without checking `ancestors`, we could get false positives for identical VirtualHosts that are located in different spots in the tree. That seems bad. If we're comparing across implementations and therefore can't look at `path` or `node`, `ancestors` seems like the only way to check for sure that they're actually the same. Although, as I type this, I realize now that that still wouldn't differentiate between sibling VirtualHosts with the same ancestors. So we may have a larger problem here.

Reply:
My hope is that someone will jump in and say, "ah, but you forgot about x, therefore this isn't actually a problem."

Reply:
That sounds right, but I'm not sure we necessarily have to use ancestors to disambiguate.What's the minimum set of information we need to disambiguate two otherwise identical VirtualHost blocks? In Augeas, the augeas path seems like it'll do that. In apacheconfig land, I think we might be able to just compare memory addresses, since the tree is generated once at the beginning. Tagging @joohoi here too.

I will also think about this a bit more.

Reply:
Yes, I agree that full path in augeas and testing if they're the same actual object in apacheconfig should do it. That does make checking across implementation impossible though.